packer/builder/azure/config.go

```diff
@@ -44,6 +44,7 @@ type Config struct {
 	tmpServiceName   string
 	tmpContainerName string
 	userImageName    string
+        SSHPrivateIp     bool   `mapstructure:"ssh_private_ip"`
```

Comment:
`ssh_private_ip` -> `use_private_ip`, also, let's do `SSHPrivateIp` -> `UsePrivateIP` to keep it consistent. (This should also made to work similarly for Windows VM's, once I get those over to using WinRM instead of CustomScript.)